lib/BackgroundJob/Tasks/CheckRequirementsTask.php

```diff
@@ -105,8 +97,37 @@ public function execute(FaceRecognitionContext $context) {
 				"If you already set this to unlimited, it seems your system is not having enough RAM memory.";
 			$this->logInfo($error_message);
 			return false;
+		} else {
+			$this->logDebug(sprintf('Found %d MB available to PHP.', $memory / (1024 * 1024)));
+			// No point going above 4GB ("4GB should be enough for everyone")
+			if ($memory > 4 * 1024 * 1024 * 1024) {
+				$this->logDebug('Capping used memory to maximum of 4GB');
+				$memory = 4 * 1024 * 1024 * 1024;
+			}
 		}
 
+		// Apply admin setting limit.
+		$memorySetting = $this->config->getAppValue('facerecognition', 'memory-limits', '-1');
+		if ($memorySetting > 0) {
+			if ($memorySetting > $memory) {
+				$error_message =
```

Comment:
I don't know about this error. I mean, if user have 2GB and configured it to 4GB, I would use 2GB and just output some warning that user limit is greater than system limit. Intent of user limit should be not to overload system above some value, but if system/PHP cap is below that, I don't think of it as a problem. In other words, limit should be `min(userLimit, memoryLimit)`. 

So, I would just output warning here, but would not block execution or error out

Reply:
Done..